src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs

```diff
@@ -553,10 +284,9 @@ private void ParseReferenceAssemblies()
                 bool nostdlib = _arguments.ContainsArgument(Options.Cmd.Nostdlib);
 
                 if (excludeTypesArgs != null && excludeTypesArgs.Count > 0)
-                    SetAllowedModesFromOption(ToolMode.ProxyGeneration, ToolMode.MetadataFromAssembly | ToolMode.ProxyGeneration | ToolMode.DataContractImport | ToolMode.XmlSerializerGeneration | ToolMode.ServiceContractGeneration, Options.Cmd.ExcludeType, null);
+                    SetAllowedModesFromOption(ToolMode.XmlSerializerGeneration, ToolMode.XmlSerializerGeneration, Options.Cmd.ExcludeType, null);
 
                 AddReferencedTypes(referencedAssembliesArgs, excludeTypesArgs, referencedCollectionTypesArgs, nostdlib);
-                _parent._typeResolver = CreateTypeResolver(_parent);
             }
```

Comment:
Why don't we need to add a type resolver any more?

Reply:
I can only see this typeresolver used in ServiceLoader in wsdlexporter.

Reply:
I of course can add it back if you feel it is more safe.

Reply:
_parent._typeResolver may only be used in ServiceLoader, but the method CreateTypeResolver also installs an AppDomain type resolver and assembly resolver so you don't need to explicitly call it for it to be used. This is the main logic for CreateTypeResolver:
```csharp
                TypeResolver typeResolver = new TypeResolver(options);
                AppDomain.CurrentDomain.TypeResolve += new ResolveEventHandler(typeResolver.ResolveType);
                AppDomain.CurrentDomain.AssemblyResolve += new ResolveEventHandler(typeResolver.ResolveAssembly);
```
If _parent._typeResolver isn't used, you could change the method to return void and not save a reference to the type resolver.

Reply:
To be more safe, I already added it back. I will review later and see what kind of change is required.